airbyte-integrations/connectors/source-file/source_file/source.py

```diff
@@ -105,6 +106,26 @@ def _validate_and_transform(config: Mapping[str, Any]):
             raise AirbyteTracedException(message=message, internal_message=message, failure_type=FailureType.config_error)
         return config
 
+    def spec(self, logger: AirbyteLogger) -> ConnectorSpecification:
+        """Returns the json schema for the spec"""
+        spec = super().spec(logger)
+
+        # override cloud spec to remove local file support
+        if is_cloud_environment():
+            for provider in spec.connectionSpecification["properties"]["provider"]["oneOf"]:
+                storage = provider["properties"]["storage"]
+
+            if "enum" in storage:
```

Comment:
📚 comments as to what we are doing could be useful

Reply:
particularly for 
>                 storage["const"] = storage.pop("default")

Reply:
Good catch - I tried to write a comment but realized I didn't understand what this was doing (I copied it from source-file-secure). Seems like when it was introduced, the spec for source-file had `enum`s everywhere and it should have been `const`s and for some reason it was patched here for source-file-secure. This is no longer the case and source-file now properly declares these as `const`, so this is not needed.

Removed the code!